src/gui_common/CustomRichTextLabel.cs

```diff
@@ -0,0 +1,245 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Godot;
+
+/// <summary>
+///   For extra functionality added on top of normal RichTextLabel. Includes custom bbcode parser.
+/// </summary>
+public class CustomRichTextLabel : RichTextLabel
+{
+    private string extendedBbcode;
+
+    /// <summary>
+    ///   Custom Bbcode tags exclusive for Thrive. Acts more like an extension to the built-in tags.
+    /// </summary>
+    public enum ThriveBbCodeTag
+    {
+        /// <summary>
+        ///   Turns compound internal name string into a bolded display name and an icon next to it.
+        ///   Mainly used in organelle tooltips.
+        /// </summary>
+        Compound,
+
+        /// <summary>
+        ///   Turns input action string into a key prompt image of its primary input event.
+        /// </summary>
+        Input,
+    }
+
+    /// <summary>
+    ///   This supports custom bbcode tags specific to Thrive (for example: [thrive:compound]glucose[/thrive:compound])
+    /// </summary>
+    /// <remarks>
+    ///   <para>
+    ///     NOTE: including "thrive" namespace in the tag is a must, otherwise the custom parser wouldn't parse it.
+    ///   </para>
+    /// </remarks>
+    [Export]
+    public string ExtendedBbcode
+    {
+        get => extendedBbcode;
+        set
+        {
+            if (extendedBbcode == value)
+                return;
+
+            extendedBbcode = value;
+
+            // Need to delay this so we can get the correct input controls from settings.
+            Invoke.Instance.Queue(() => ParseCustomTags());
+        }
+    }
+
+    public override void _Ready()
+    {
+        // Make sure bbcode is enabled
+        BbcodeEnabled = true;
+    }
+
+    /// <summary>
+    ///   Parses a custom tagged substring into a templated bbcode.
+    /// </summary>
+    private void ParseCustomTags()
+    {
+        var result = new StringBuilder();
+        var currentTagBlock = new StringBuilder();
+
+        var tagStack = new Stack<string>();
+
+        var isIteratingTag = false;
+        var isIteratingContent = false;
+
+        // The index of a closing bracket in a last iterated opening tag, used
+        // to retrieve the tagged substring
+        var lastStartingTagEndIndex = 0;
+
+        for (int index = 0; index < ExtendedBbcode.Length; index++)
```

Comment:
I think just for safety regarding performance, I'd prefer if this directly read the object field here instead of going through the field (as that incurs an extra method call in C# and we definitely want this parser to be fast)